exec/java-exec/src/main/java/org/apache/drill/exec/store/pcapng/PcapColumn.java

```diff
@@ -0,0 +1,1031 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.store.pcapng;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.apache.drill.common.types.TypeProtos.MajorType;
+import org.apache.drill.common.types.TypeProtos.MinorType;
+import org.apache.drill.common.types.Types;
+import org.apache.drill.exec.store.pcap.PcapFormatUtils;
+import org.apache.drill.exec.store.pcapng.decoder.PacketDecoder;
+import org.apache.drill.exec.vector.accessor.ScalarWriter;
+import org.joda.time.Instant;
+
+import fr.bmartel.pcapdecoder.structure.options.inter.IOptionsStatisticsHeader;
+import fr.bmartel.pcapdecoder.structure.types.IPcapngType;
+import fr.bmartel.pcapdecoder.structure.types.inter.IDescriptionBlock;
+import fr.bmartel.pcapdecoder.structure.types.inter.IEnhancedPacketBLock;
+import fr.bmartel.pcapdecoder.structure.types.inter.INameResolutionBlock;
+import fr.bmartel.pcapdecoder.structure.types.inter.ISectionHeaderBlock;
+import fr.bmartel.pcapdecoder.structure.types.inter.IStatisticsBlock;
+
+public abstract class PcapColumn {
+
+  private static final Map<String, PcapColumn> columns = new LinkedHashMap<>();
+  private static final Map<String, PcapColumn> summary_columns = new LinkedHashMap<>();
+  public static final String DUMMY_NAME = "dummy";
+  public static final String PATH_NAME = "path";
+
+  static {
+    // Basic
+    columns.put("timestamp", new PcapTimestamp());
+    columns.put("packet_length", new PcapPacketLength());
+    columns.put("type", new PcapType());
+    columns.put("src_ip", new PcapSrcIp());
+    columns.put("dst_ip", new PcapDstIp());
+    columns.put("src_port", new PcapSrcPort());
+    columns.put("dst_port", new PcapDstPort());
+    columns.put("src_mac_address", new PcapSrcMac());
+    columns.put("dst_mac_address", new PcapDstMac());
+    columns.put("tcp_session", new PcapTcpSession());
+    columns.put("tcp_ack", new PcapTcpAck());
+    columns.put("tcp_flags", new PcapTcpFlags());
+    columns.put("tcp_flags_ns", new PcapTcpFlagsNs());
+    columns.put("tcp_flags_cwr", new PcapTcpFlagsCwr());
+    columns.put("tcp_flags_ece", new PcapTcpFlagsEce());
+    columns.put("tcp_flags_ece_ecn_capable", new PcapTcpFlagsEceEcnCapable());
+    columns.put("tcp_flags_ece_congestion_experienced", new PcapTcpFlagsEceCongestionExperienced());
+    columns.put("tcp_flags_urg", new PcapTcpFlagsUrg());
+    columns.put("tcp_flags_ack", new PcapTcpFlagsAck());
+    columns.put("tcp_flags_psh", new PcapTcpFlagsPsh());
+    columns.put("tcp_flags_rst", new PcapTcpFlagsRst());
+    columns.put("tcp_flags_syn", new PcapTcpFlagsSyn());
+    columns.put("tcp_flags_fin", new PcapTcpFlagsFin());
+    columns.put("tcp_parsed_flags", new PcapTcpParsedFlags());
+    columns.put("packet_data", new PcapPacketData());
```

Comment:
@luocooong 
I saw your response.  Basically, what we did in the original reader was put a lot of code in try/catch blocks and if it found errors, set `is_corrupt` to `true` and move on to the next packet.  Have you tested the PCAP-NG with a corrupt packet?  Does the `parse()` method throw an exception if the packet is corrupt?
The reason I'm asking is that it is often better for the user to get a result, rather than a huge java stacktrace, so if we can avoid that, it would be good.